### PR TITLE
KEH-317 | Cognito Authentication

### PR DESCRIPTION
**Pull request by TotalDwarf03**
Changes include:
- New Terraform module `authentication/main.tf`
  - This creates an AWS Cognito user pool to control access to the dashboard
- Update to dashboard terraform so that `dashboard/alb.tf` now redirects the user through Cognito before giving access to the tool.
- README changes to provide instructions (I don't like the READMEs at the moment - they feel messy)

Also includes some UI changes to remove Secret value from frontend and dependabot advisory (the CAB will probably fail the dashboard on this).

### CODE REVIEW

File: terraform/dashboard/alb.tf [hunk 1]
```diff
@@ -29,17 +29,27 @@ resource "aws_lb_listener_rule" "service_listener_rule" {
     }
   }
 
-  dynamic "action" {
-    for_each = var.authenticate_users ? [1] : []
+  # dynamic "action" {
```
Comment (gibbardsteve) [inline terraform/dashboard/alb.tf:32]: I think the idea of the dynamic action was to allow you to enable the auth by setting the authenticate_users variable to 1.  Did you try this?  If you are not using the dynamic element, it'd be best delete the commented code and run terraform fmt to ensure the code is formatted correctly.

**Review by gibbardsteve — COMMENTED** [review]
Minor - otherwise looks ok

File: terraform/dashboard/alb.tf (hunk 1 above)
Comment (TotalDwarf03) [reply terraform/dashboard/alb.tf:32]: Keeping to static action to match Repo Archive Tool

File: src/app.py [hunk 2]
```diff
@@ -574,7 +585,7 @@ def load_file(filename: str) -> dict:
                     [
                         "Repository Name",
                         "Dependency",
-                        "Advisory",
```
Comment (gibbardsteve) [inline src/app.py:577]: Did you mean for this change to come in?

**Review by gibbardsteve — APPROVED** [review]
Check the comment before merge.

File: src/app.py (hunk 2 above)
Comment (TotalDwarf03) [reply src/app.py:577]: Yes, it removes information from the dashboard that the CAB probably won't like. Did it in this PR so we can give the dashboard back to CAB.

**Pull request merged**